core/src/main/scala/org/apache/spark/ui/ConsoleProgressBar.scala

```diff
@@ -32,9 +32,14 @@ private[spark] class ConsoleProgressBar(sc: SparkContext) extends Logging {
   // Carriage return
   val CR = '\r'
   // Update period of progress bar, in milliseconds
-  val UPDATE_PERIOD = 200L
+  val UPDATE_PERIOD_MSEC = if (SparkEnv.get == null) {
```

Comment:
Can this be null, and, can't you use the context's conf here instead?
These can be `private` perhaps?
These constants really should have been in an `object`, but now that this sorta needs to be a field anyway, consider making its name more like a field name "updatePeriodMSec"